Library/Homebrew/compat/macos.rb

```diff
@@ -2,97 +2,170 @@
+
+      def lion_or_newer?
+        opoo "Calling MacOS.lion_or_newer? is deprecated!"
+        puts "Use 'MacOS.version >= :lion' instead."
+        version >= :lion
+      end
+
+      def mountain_lion?
+        opoo "Calling MacOS.mountain_lion? is deprecated!"
+        puts "Use 'MacOS.version >= :mountain_lion' instead."
+        version >= :mountain_lion
+      end
+
+      def mountain_lion_or_newer?
+        opoo "Calling MacOS.mountain_lion_or_newer? is deprecated!"
+        puts "Use 'MacOS.version >= :mountain_lion' instead."
+        version >= :mountain_lion
+      end
+
+      def macports_or_fink_installed?
+        opoo "Calling MacOS.macports_or_fink_installed? is deprecated!"
+        puts "Use !MacOS.macports_or_fink.empty? instead."
+        !macports_or_fink.empty?
+      end
+
+      def locate(tool)
+        opoo "Calling MacOS.locate is deprecated!"
+        puts "Use DevelopmentTools.locate instead."
+        DevelopmentTools.locate(tool)
+      end
+
+      def default_cc
+        opoo "Calling MacOS.default_cc is deprecated!"
+        puts "Use DevelopmentTools.default_cc instead."
+        DevelopmentTools.default_cc
+      end
+
+      def default_compiler
+        opoo "Calling MacOS.default_compiler is deprecated!"
+        puts "Use DevelopmentTools.default_compiler instead."
+        DevelopmentTools.default_compiler
+      end
+
+      def gcc_40_build_version
+        opoo "Calling MacOS.gcc_40_build_version is deprecated!"
+        puts "Use DevelopmentTools.gcc_40_build_version instead."
+        DevelopmentTools.gcc_40_build_version
+      end
+
+      def gcc_4_0_build_version
+        opoo "Calling MacOS.gcc_4_0_build_version is deprecated!"
+        puts "Use DevelopmentTools.gcc_40_build_version instead."
+        DevelopmentTools.gcc_40_build_version
+      end
+
+      def gcc_42_build_version
+        opoo "Calling MacOS.gcc_42_build_version is deprecated!"
+        puts "Use DevelopmentTools.gcc_42_build_version instead."
+        DevelopmentTools.gcc_42_build_version
+      end
+
+      def gcc_build_version
+        opoo "Calling MacOS.gcc_build_version is deprecated!"
+        puts "Use DevelopmentTools.gcc_42_build_version instead."
+        DevelopmentTools.gcc_42_build_version
+      end
+
+      def llvm_build_version
+        opoo "Calling MacOS.llvm_build_version is deprecated!"
+        puts "There is no replacement."
+      end
+
+      def clang_version
+        opoo "Calling MacOS.clang_version is deprecated!"
+        puts "Use DevelopmentTools.clang_version instead."
+        DevelopmentTools.clang_version
+      end
+
+      def clang_build_version
+        opoo "Calling MacOS.clang_build_version is deprecated!"
+        puts "Use DevelopmentTools.clang_build_version instead."
+        DevelopmentTools.clang_build_version
+      end
+
+      # We won't change the name because of backward compatibility.
+      # So disable rubocop here.
+      # rubocop:disable Style/PredicateName
```

Comment:
> Ideally we'd disable this for all of `compat/`.

That would be fine by me and I would either prefer that or prefer to make the effect of those comments more fine-grained by moving them to the end of the respective `def` lines. The former can be achieved by adding a `.rubocop.yml` into the `compat/` directory and disabling the annoying rules in there, e.g.:

```yaml
inherit_from: ../../.rubocop.yml

Style/MethodName:
  Enabled: false
Style/PredicateName:
  Enabled: false
```

This works well with editor integration and invoking RuboCop directly, because it starts the search for such a file in the directory of the file that is being inspected. `brew style` defeats this logic by directly providing a configuration file, so it will still complain and ignore the extra `.rubocop.yml` file. (I think we should adjust `brew style` not to do that.)

(Naturally this extends to the other related code comments.)